### DISCUSSION TITLE
GSIs that share the hash with the primary index

### DISCUSSION THREAD

**Discussion by dibell** [sensedeep/dynamodb-onetable#277] [Q&A] [answered]
This is more of a question that an issue... 
(using v2.3.1 currently)

Why do indexes get created as LSIs if hash is the same as the primary hash even though local is not set to true?
E.g. 
```
    indexes: {
        primary: { hash: 'pk', sort: 'sk'},
        gsi0: { hash: 'pk', sort: 'anotherProp', local: false},
```
results in an LSI

I have a use case where I want to use several GSIs that have the same hash as the primary (mostly to avoid the limitations of LSIs)
This can be achieved by duplicating the primary hash into a unique property for each GSI but this is unnecessary duplication so I was wondering why this choice was made, if it could be changed or if there's a better way to do this?

Many thanks

**Comment by mobsense** [comment]
The doc is not correct in the examples:

You should use 

```
    indexes: {
        primary: { hash: 'pk', sort: 'sk'},
        gsi0: { hash: 'pk', sort: 'anotherProp', type: 'local'},
    }
```

**Reply by dibell** [reply]
But I need the gsi to be global not local.
I've found the code that does this in src/Schema.js
```
            if (name != 'primary') {
                if (index.type == 'local') {
                ...
                } else if (index.hash == primary.hash) {
                    index.type = 'local'
```
Why does it need to do this?

**Comment by mobsense** [comment]
I'll move this to discussions.

**Comment by mobsense** [comment, marked as answer]
All indexes are GSIs by default (except the primary). If you have no `type` in the index definition, it will be a GSI.
